go/parquet/metadata/app_version.go

```diff
@@ -164,7 +164,7 @@ func (v AppVersion) HasCorrectStatistics(coltype parquet.Type, logicalType schem
 	// parquet-cpp-arrow version 4.0.0 fixed Decimal comparisons for creating min/max stats
 	// parquet-cpp also becomes parquet-cpp-arrow as of version 4.0.0
 	if v.App == "parquet-cpp" || (v.App == "parquet-cpp-arrow" && v.LessThan(parquet1655FixedVersion)) {
-		if _, ok := logicalType.(*schema.DecimalLogicalType); ok && coltype == parquet.Types.FixedLenByteArray {
+		if _, ok := logicalType.(schema.DecimalLogicalType); ok && coltype == parquet.Types.FixedLenByteArray {
```

Comment:
why did this change?

Reply:
Some implementations of `LogicalType` were using value receivers and others were using pointer receivers, which made it fairly inconsistent and error-prone to work with them. Changing them all to use the same convention felt easier to work with, and since none of the interface methods mutate the struct (i.e. data-only) I went with a value receiver.